### PR TITLE
Feature/foundation table font sizes

### PR DESCRIPTION
**Pull request by ulferts**
Addresses the table part of https://community.openproject.org/work_packages/18365

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://feature-foundation-table-font-sizes.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://feature-foundation-table-font-sizes.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by ulferts** [comment]
@myabc can you please have a look at this and tell me if this goes into the right direction?

I haven't been able to remove the typograpy from foundation as that causes a lot of regressions. 

I still have to see the benefits of applying styles in dom vs. having css inheritance. I had to change a whole lot of code at a whole lot of places when I could have simply extended `table.list tbody` with the right font-size. I still might do this to take care of the plugins. On the other hand. The difference is not that huge so we might get away with not styling the tables in the plugins :).

File: app/assets/stylesheets/default.css.sass [hunk 1]
```diff
@@ -34,9 +34,11 @@
 
 @import open_project_global/all
 
-@import foundation
+@import content/typography
```
Comment (myabc) [inline app/assets/stylesheets/default.css.sass:37]: @ulferts the content partial should be loaded after the framework. If we need to make the variables available to `_settings.scss`, then move them into our own `open_project_global/_variables.sass`.

File: app/views/layouts/angular.html.erb [hunk 2]
```diff
@@ -54,7 +54,7 @@ See doc/COPYRIGHT.rdoc for more details.
     <!-- page specific tags -->
     <%= content_for(:header_tags) if content_for?(:header_tags) %>
   </head>
-  <body class="<%= body_css_classes %>" ng-app="openproject" data-relative_url_root="<%= root_path %>">
+  <body class="font-medium <%= body_css_classes %>" ng-app="openproject" data-relative_url_root="<%= root_path %>">
```
Comment (myabc) [inline app/views/layouts/angular.html.erb:57]: @ulferts why is this class on `body`?

**Comment by myabc** [comment]
@ulferts 

> can you please have a look at this and tell me if this goes into the right direction?
> 
> I had to change a whole lot of code at a whole lot of places when I could have simply extended table.list tbody with the right font-size.

Applying font classes to the body in this way isn't really what I envisaged (and I agree with you, that it makes for a rather cumbersome approach). As far as an alternative suggestion is concerned though, I still need to think it through… :frowning:

**Comment by ulferts** [comment]
@myabc I have no problem with specifying it on the body once (or at least once for every layout). It defines very expressively that the font-size in that area is small. One could of course place it on the content class as well but that can be done once we define different font sizes for the other regions like menu and header. 

What I struggle with is that now everybody has to remember to set the font-size to small in the tbody. That is very error prone and also makes the application hard to change.

That problem would of course not exist if we where to use a table builder which would set the appropriate class for all the tables. However, this is out of scope.

**Comment by floriank** [comment]
#2758 will address this

**Pull request closed without merging**